### PR TITLE
Resolve note

### PR DESCRIPTION
**Pull request by Azmaa22**

### CODE REVIEW

**Comment by Azmaa22** [comment]
Hello @TarekAlabd , I Tried resolving the notes you told me and resolve conflicts after pull and merging so please before you run project clean it to avoid any problem may be occurred

File: lib/main.dart [hunk 1]
```diff
@@ -17,50 +20,20 @@ class MyApp extends StatelessWidget {
...
-                ),
-              ),
-              focusedErrorBorder: OutlineInputBorder(
-                borderRadius: BorderRadius.circular(2.0),
-                borderSide: const BorderSide(
-                  color: Colors.red,
-                ),
-              ),
-            )),
-        onGenerateRoute: onGenerate,
```
Comment (TarekAlabd) [inline lib/main.dart:61]: Why removing this?

File: lib/utilities/constants/theme_manager.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        primary: ColorManager.kRed,
+        shadowColor: ColorManager.kBlack,
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(
+            25.0,
+          ),
+        ),
+      ),
+    ),
+    textTheme: const TextTheme(
```
Comment (TarekAlabd) [inline lib/utilities/constants/theme_manager.dart:60]: Why did you change the values of the theme itself?
I think we can use the default values as the material set.

File: lib/utilities/router.dart [hunk 3]
```diff
@@ -2,12 +2,15 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter_ecommerce/utilities/routes.dart';
 import 'package:flutter_ecommerce/views/pages/bottom_navbar.dart';
 import 'package:flutter_ecommerce/views/pages/landing_page.dart';
-import 'package:flutter_ecommerce/views/pages/auth_page.dart';
+import 'package:flutter_ecommerce/views/pages/screens.dart';
 
 Route<dynamic> onGenerate(RouteSettings settings) {
   switch (settings.name) {
-    case AppRoutes.loginPageRoute:
+    case AppRoutes.navBottomContainerPageRoute:
```
Comment (TarekAlabd) [inline lib/utilities/router.dart:9]: Why adding this route and there's one exists
You can refactor depend on it, just put the other things in separate widgets

File: lib/utilities/routes.dart [hunk 4]
```diff
@@ -2,5 +2,10 @@ class AppRoutes {
   static const String landingPageRoute = '/';
   static const String loginPageRoute = '/login';
   static const String registerPageRoute = '/register';
+
+
```
Comment (TarekAlabd) [inline lib/utilities/routes.dart:6]: There's an error in this file and avoid duplication

File: lib/views/pages/home_page/widgets/home_header.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            ),
+            Padding(
+              padding: EdgeInsets.only(
+                bottom: size.height * 0.03,
+              ),
+              child: InkWell(
+                onTap: viewAll,
+                child: const Text(
+                  'View all',
+                  style: TextStyle(
```
Comment (TarekAlabd) [inline lib/views/pages/home_page/widgets/home_header.dart:47]: Use the text themes themselves from Theme.of(context)

File: lib/views/pages/nav_bottom_container/nav_bottom_container.dart [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_ecommerce/views/pages/nav_bottom_container/widgets/nav_bottom_items.dart';
+import 'package:flutter_ecommerce/views/pages/nav_bottom_container/widgets/nav_bottom_screens.dart';
+import 'package:persistent_bottom_nav_bar/persistent-tab-view.dart';
+
+class NavigationBottomContainer extends StatefulWidget {
```
Comment (TarekAlabd) [inline lib/views/pages/nav_bottom_container/nav_bottom_container.dart:6]: This is a duplicated widget.
We can do this refactor on the origin widget and separate the other things in separated widget if possible

File: lib/views/widgets/input.dart [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      autofocus: true,
+      focusNode: focusNode,
+      textInputAction: textInputAction,
+      onEditingComplete: onEditingComplete,
+      enabled: isReadOnly,
+      initialValue: initialValue,
+      onChanged: onChange,
+      validator: onValidate,
+      name: name,
+      style: TextStyle(
```
Comment (TarekAlabd) [inline lib/views/widgets/input.dart:50]: Use the theme instead of TextStyle

File: lib/views/widgets/social_button.dart [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+          child: SvgPicture.asset(
+            iconName,
+            width: 40.0,
+            height: 40.0,
+          ),
+        ),
+        width: 80.0,
+        height: 80.0,
+        decoration: BoxDecoration(
+          color: ColorManager.kWhite,
```
Comment (TarekAlabd) [inline lib/views/widgets/social_button.dart:28]: Use the colors from the theme instead of that

File: pubspec.yaml [hunk 9]
```diff
@@ -2,28 +2,54 @@ name: flutter_ecommerce
...
 
+
 flutter:
+
   uses-material-design: true
+
+  
   assets:
-     - assets/
+    - assets/images/
```
Comment (TarekAlabd) [inline pubspec.yaml:44]: This file is not exist, so we don't need to add it for now

File: pubspec.yaml [hunk 10]
```diff
@@ -2,28 +2,54 @@ name: flutter_ecommerce
...
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.17.0 <3.0.0"
+  sdk: ">=2.16.1 <3.0.0"
 
 dependencies:
   flutter:
     sdk: flutter
+
```
Comment (TarekAlabd) [inline pubspec.yaml:14]: Try to remove the white spaces between the dependencies
Just one line between the different headlines and between the dependencies themselves: no space.

**Review by TarekAlabd — CHANGES_REQUESTED** [review]
Hello Asmaa
I hope you are doing great
Again, I really appreciate your efforts and really you're doing amazing
I just have some comments you can work on quickly I think and I will review them again and merge them insha'Allah.

One additional note:
We need to use the same architecture, so we don't need to add the 'data' folder at all and inside the pages, we can add the page inside it without adding a new directory.

MVC: Model View Controller
We can refactor and change this architecture in the future insha'Allah to one better.

File: lib/views/widgets/social_button.dart (hunk 8 above)
Comment (Azmaa22) [reply lib/views/widgets/social_button.dart:28]: How can I do ? @TarekAlabd
